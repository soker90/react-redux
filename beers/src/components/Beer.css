.beer {
  position: relative;
  padding: 15px;
}

.beer-error {
  color: red;
}
