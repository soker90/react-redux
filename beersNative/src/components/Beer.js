import React, { Component } from "react"
import BeerForm from "./BeerForm"
import BeerList from "./BeerList"
import { Text, View } from 'react-native'

class Beer extends Component {
  constructor(props) {
    super(props)

    this.onRemove = this.onRemove.bind(this)

  }

  componentDidMount() {
    this.props.fetchBeer()
  }

  onRemove(beerId) {
	  this.props.removeBeer(beerId)
  }

  render() {
    const {beerList, error, loading} = this.props
    return (
      <View className="beer">

        <BeerForm addBeer={this.props.addBeer} loading={loading}/>
        {error && <Text className="beer-error"> An beer error occurred: {error.statusText}: {error.status}</Text>}
        {beerList.length
          ? <BeerList beerList={beerList} removeBeer={this.onRemove} loading={loading}/>
          : <Text>The beer list is empty</Text>}
      </View>
    )
  }

}

export default Beer
