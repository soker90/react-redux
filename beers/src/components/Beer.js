import React, { Component } from "react"
import BeerForm from "./BeerForm"
import BeerList from "./BeerList"
import "./Beer.css"

class Beer extends Component {
  constructor(props) {
    super(props)

    this.onRemove = this.onRemove.bind(this)

  }

  componentDidMount() {
    this.props.fetchBeer()
  }

  onRemove(beerId) {
    let modalData = {
      title: 'Are you sure?',
      body: 'You want to delete the beer with id: ' + beerId,
      callback: this.props.removeBeer,
      extraParams: beerId
    }

    this.props.openModal(modalData)
  }

  render() {
    const {beerList, error, loading} = this.props

    return (
      <div className="beer">

        <BeerForm addBeer={this.props.addBeer} loading={loading}/>
        {error && <p className="beer-error"> An beer error occurred: {error.statusText}: {error.status}</p>}
        {beerList.length
          ? <BeerList beerList={beerList} removeBeer={this.onRemove} loading={loading}/>
          : <p>The beer list is empty</p>}
      </div>
    )
  }

}

export default Beer
