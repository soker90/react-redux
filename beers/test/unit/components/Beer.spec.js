// Beer.spec.js
import React from 'react'
import renderer from 'react-test-renderer'

import Beer from '../../../src/components/Beer'
import mockData from '../mockData'

describe('Beer component', () => {
  test('render Beer component without crashing', () => {
    const beerList = []
    const loading = false
    const error = null
    const fetchBeer = jest.fn()

    const tree = renderer.create(
      <Beer beerList={beerList} loading={loading} error={error} fetchBeer={fetchBeer}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render Beer component with beers', () => {
    const beerList = mockData.beers
    const loading = false
    const error = null
    const fetchBeer = jest.fn()

    const tree = renderer.create(
      <Beer beerList={beerList} loading={loading} error={error} fetchBeer={fetchBeer}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render Beer component with loading true', () => {
    const beerList = []
    const loading = true
    const error = null
    const fetchBeer = jest.fn()

    const tree = renderer.create(
      <Beer beerList={beerList} loading={loading} error={error} fetchBeer={fetchBeer}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

  test('render Beer component with error', () => {
    const beerList = []
    const loading = false
    const error = {
      status: 401,
      statusText: 'Unauthorized'
    }
    const fetchBeer = jest.fn()

    const tree = renderer.create(
      <Beer beerList={beerList} loading={loading} error={error} fetchBeer={fetchBeer}/>
    )
    expect(tree.toJSON()).toMatchSnapshot()
  })

})
