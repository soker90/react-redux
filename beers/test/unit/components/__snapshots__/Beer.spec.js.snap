exports[`Beer component render Beer component with beers 1`] = `
<div
  className="beer">
  <div>
    <form
      action="#"
      onSubmit={[Function]}>
      <input
        required="required"
        type="text" />
      <input
        required="required"
        type="text" />
      <input
        disabled={false}
        type="submit"
        value="Add Beer" />
    </form>
  </div>
  <div>
    <div>
      <p>
        <b>
          Amstel
        </b>
         (
        5
        %) 
        <i>
          beer_01
        </i>
      </p>
      <div>
        <button
          disabled={false}
          onClick={[Function]}>
          Remove Beer
        </button>
      </div>
    </div>
    <div>
      <p>
        <b>
          Mahoo
        </b>
         (
        3
        %) 
        <i>
          beer_02
        </i>
      </p>
      <div>
        <button
          disabled={false}
          onClick={[Function]}>
          Remove Beer
        </button>
      </div>
    </div>
    <div>
      <p>
        <b>
          Corona
        </b>
         (
        4,5
        %) 
        <i>
          beer_03
        </i>
      </p>
      <div>
        <button
          disabled={false}
          onClick={[Function]}>
          Remove Beer
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Beer component render Beer component with error 1`] = `
<div
  className="beer">
  <div>
    <form
      action="#"
      onSubmit={[Function]}>
      <input
        required="required"
        type="text" />
      <input
        required="required"
        type="text" />
      <input
        disabled={false}
        type="submit"
        value="Add Beer" />
    </form>
  </div>
  <p
    className="beer-error">
     An beer error occurred: 
    Unauthorized
    : 
    401
  </p>
  <p>
    The beer list is empty
  </p>
</div>
`;

exports[`Beer component render Beer component with loading true 1`] = `
<div
  className="beer">
  <div>
    <form
      action="#"
      onSubmit={[Function]}>
      <input
        required="required"
        type="text" />
      <input
        required="required"
        type="text" />
      <input
        disabled={true}
        type="submit"
        value="Loading..." />
    </form>
  </div>
  <p>
    The beer list is empty
  </p>
</div>
`;

exports[`Beer component render Beer component without crashing 1`] = `
<div
  className="beer">
  <div>
    <form
      action="#"
      onSubmit={[Function]}>
      <input
        required="required"
        type="text" />
      <input
        required="required"
        type="text" />
      <input
        disabled={false}
        type="submit"
        value="Add Beer" />
    </form>
  </div>
  <p>
    The beer list is empty
  </p>
</div>
`;
